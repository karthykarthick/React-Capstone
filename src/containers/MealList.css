.meals-container {
  margin-bottom: 50px;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

.meals {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.add-button {
  padding: 10px 30px;
  background-color: #ff772e;
  border: none;
  color: white;
  cursor: pointer;
  font-weight: bold;
  outline: none;
  transition: transform, 0.2s ease-in-out;
}

.add-button:hover {
  transform: scale(1.1);
}

.warning {
  opacity: 0.6;
  margin-top: 40px;
}

.meal {
  display: flex;
  width: 235px;
  padding: 5px 5px;
  padding-right: 10px;
  box-shadow: 0 0 15px -5px rgba(0, 0, 0, 0.473);
  border-radius: 5px;
  margin-bottom: 30px;
  text-decoration: none;
  color: black;
}

.hidden {
  visibility: hidden;
}

.meal__image {
  width: 100px;
  height: 100px;
  margin-right: 10px;
}

.meal__info {
  padding: 5px 0;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
}

.meal__info__title {
  font-size: 14px;
}

.meal__info__category {
  align-self: flex-end;
  opacity: 0.7;
  font-size: 12px;
}

@media screen and (max-width: 425px) {
  .meal {
    width: 100%;
  }

  .meal__image {
    width: 120px;
    height: 120px;
  }
}

@media screen and (max-width: 768px) and (min-width: 424px) {
  .meal {
    width: 335px;
  }
}
