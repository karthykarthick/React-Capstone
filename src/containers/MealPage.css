.heading {
  margin-bottom: 20px;
}

.title {
  color: rgba(0, 0, 0, 0.856);
  font-family: Arial, Helvetica, sans-serif;
  font-size: 40px;
  margin-bottom: 5px;
  line-height: 1;
}

.area {
  opacity: 0.5;
  font-size: 14px;
}

.video-container {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin-bottom: 30px;
}

.video {
  width: 700px;
  border: none;
  height: calc(700px / 1.78);
}

.headline {
  font-weight: bold;
  font-size: 30px;
  margin-top: 30px;
}

.preparation {
  display: flex;
  justify-content: center;
  align-items: center;
}

.image {
  width: 400px;
  margin-right: 30px;
  object-fit: cover;
}

.ingredients h4 {
  font-size: 30px;
  margin-bottom: 10px;
  line-height: 1;
}

.ingredients ul {
  list-style: circle inside;
}

.ingredients li {
  font-size: 14px;
  opacity: 0.8;
}

.instructions {
  max-width: 650px;
  text-align: justify;
  margin: 30px auto;
}

.instructions p {
  font-size: 14px;
}

@media screen and (max-width: 425px) {
  .video {
    width: calc(100vw - 60px);
    height: calc((100vw - 20px) / 1.78);
  }

  .preparation {
    flex-direction: column;
    align-items: flex-start;
  }

  .preparation .image {
    width: 100%;
    margin-right: 0;
    margin-bottom: 10px;
  }

  .headline {
    font-size: 13.5px;
  }

  .heading .title {
    font-size: 30px;
  }
}
