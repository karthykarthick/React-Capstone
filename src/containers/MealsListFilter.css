.filter {
  margin-bottom: 30px;
  display: flex;
}

.filter__input {
  width: 400px;
}

.filter__selector {
  width: 150px;
}

.filter__input,
.filter__selector,
.filter__button {
  padding: 5px 10px;
  font-size: 16px;
  border: 1px solid rgba(0, 0, 0, 0.199);
  border-radius: 5px;
  margin-right: 10px;
  background-color: white;
}

.filter__button {
  width: 120px;
  cursor: pointer;
}

@media screen and (max-width: 425px) {
  .filter {
    flex-direction: column;
    margin-bottom: 20px;
  }

  .filter > * {
    margin-bottom: 10px;
  }

  .filter__input {
    width: 100%;
  }
}
