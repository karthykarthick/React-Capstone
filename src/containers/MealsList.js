import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Loader from '../components/Loader';
import { startSetMeals } from '../actions/meals';
import { filterMeals } from '../helpers';
import MealListItem from '../components/MealListItem';
import './MealList.css';

export const MealsList = ({ meals, startSetMeals }) => {
  const [isLoading, setIsLoading] = useState(meals.length === 0);

  const getTenMeals = (index = 0) => {
    const result = [];
    let i = index;

    while (i < meals.length && i < index + 12) {
      result.push(meals[i]);
      i += 1;
    }

    return result;
  };

  const [localMeals, setLocalMeals] = useState([]);
  const [lastIndex, setLastIndex] = useState(0);

  const handleClick = () => {
    setLocalMeals(
      prevState => [...prevState, ...getTenMeals(lastIndex)],
    );
    setLastIndex(prevIndex => prevIndex + 12);
  };

  useEffect(() => {
    if (meals.length === 0) {
      setIsLoading(true);
      startSetMeals()
        .then(() => {
          setIsLoading(false);
        });
    }
  }, []);

  useEffect(() => {
    setLocalMeals(getTenMeals(0));
    setLastIndex(12);
  }, [meals]);

  return (
    <div className="content-container">
      {
        isLoading
          ? <Loader />
          : (
            <div className="meals-container">
              <div className="meals">
                {
                  localMeals.map(meal => (
                    <MealListItem
                      key={meal.id}
                      meal={meal}
                    />
                  ))
                }
                {
                  (localMeals.length + 1) % 3 === 0
                  && (<div className="meal hidden" />)
                }
              </div>
              {
                meals.length > localMeals.length
                  ? (
                    <button
                      type="button"
                      onClick={handleClick}
                      className="add-button"
                    >
                      Load More
                    </button>
                  )
                  : (
                    <p className="warning">No More Meals</p>
                  )
              }
            </div>
          )
      }
    </div>
  );
};

MealsList.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  startSetMeals: PropTypes.func.isRequired,
};

const mapStateToProps = ({ meals, filters }) => ({
  meals: filterMeals(meals, filters),
});

const mapDispatchToProps = { startSetMeals };

export default connect(mapStateToProps, mapDispatchToProps)(MealsList);
