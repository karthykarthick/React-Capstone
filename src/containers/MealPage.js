import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { startSetMealDetails, emptyMealDetails } from '../actions/mealDetails';
import './MealPage.css';

export const MealPage = ({
  match, startSetMealDetails, mealDetails: {
    title, area, youtubeVideo, instructions, image, ingredients,
  },
}) => {
  useEffect(() => {
    startSetMealDetails(match.params.id);

    return () => {
      emptyMealDetails();
    };
  }, []);

  return (
    <div className="content-container page-container">
      <div
        className="heading"
      >
        <h1
          className="title"
        >
          {title}
        </h1>
        <p
          className="area"
        >
          {`${area} Food`}
        </p>
      </div>
      {
        youtubeVideo
        && (
          <div className="video-container">
            <iframe
              className="video"
              title={title}
              src={youtubeVideo}
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              allowFullScreen
            />
            <div className="headline">
              {/* eslint-disable-next-line jsx-a11y/accessible-emoji */}
              ðŸ‘‡ ðŸ‘‡ ðŸ‘‡ Instructions below ðŸ‘‡ ðŸ‘‡ ðŸ‘‡
            </div>
          </div>
        )
      }
      <div
        className="preparation"
      >
        <img
          className="image"
          src={image}
          alt={`${title} thumbnail`}
        />
        <div
          className="ingredients"
        >
          <h4>Ingredients</h4>
          <ul>
            {
              ingredients.map(item => (
                <li
                  key={item}
                  className="ingredients__item"
                >
                  {item}
                </li>
              ))
            }
          </ul>
        </div>
      </div>
      <div
        className="instructions"
      >
        <h1>Instructions</h1>
        <p>
          {instructions}
        </p>
      </div>
    </div>
  );
};

MealPage.propTypes = {
  match: PropTypes.objectOf(PropTypes.any).isRequired,
  startSetMealDetails: PropTypes.func.isRequired,
  mealDetails: PropTypes.objectOf(PropTypes.any).isRequired,
};

const mapStateToProps = ({ mealDetails }) => ({
  mealDetails,
});

const mapDispatchToProps = { startSetMealDetails, emptyMealDetails };

export default connect(mapStateToProps, mapDispatchToProps)(MealPage);
