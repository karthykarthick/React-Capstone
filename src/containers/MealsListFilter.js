import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { setTextFilter, setCategoryFilter } from '../actions/filters';
import './MealsListFilter.css';

export const MealsListFilter = ({
  categories, text, category, setTextFilter, setCategoryFilter,
}) => {
  const handleTextChange = event => {
    setTextFilter(event.target.value);
  };

  const handleCategoryChange = event => {
    setCategoryFilter(event.target.value);
  };

  const handleClearClick = () => {
    setTextFilter('');
    setCategoryFilter('');
  };

  return (
    <div className="content-container">
      <div className="filter">
        <input
          type="text"
          placeholder="Search by Title"
          className="filter__input"
          value={text}
          onChange={handleTextChange}
        />
        <select
          onChange={handleCategoryChange}
          value={category}
          className="filter__selector"
        >
          <option value="">All</option>
          {
            categories.map(category => (
              <option key={category}>
                {category}
              </option>
            ))
          }
        </select>
        <button
          className="filter__button"
          type="button"
          onClick={handleClearClick}
        >
          Clear Filters
        </button>
      </div>
    </div>
  );
};

MealsListFilter.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
  text: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  setTextFilter: PropTypes.func.isRequired,
  setCategoryFilter: PropTypes.func.isRequired,
};

const mapStateToProps = ({ categories, filters }) => ({
  categories,
  text: filters.text,
  category: filters.category,
});

const mapDispatchToProps = { setCategoryFilter, setTextFilter };

export default connect(mapStateToProps, mapDispatchToProps)(MealsListFilter);
