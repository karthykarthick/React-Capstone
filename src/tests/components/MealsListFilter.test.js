import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import { categories } from '../fixtures';
import { MealsListFilter } from '../../containers/MealsListFilter';

const renderer = new ShallowRenderer();

it('should render MealsListFilter correctly', () => {
  renderer.render(
    <MealsListFilter
      categories={categories}
      text=""
      category=""
      setTextFilter={() => {}}
      setCategoryFilter={() => {}}
    />,
  );

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
