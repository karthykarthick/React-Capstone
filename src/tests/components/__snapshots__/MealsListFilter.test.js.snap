// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MealsListFilter correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="filter"
  >
    <input
      className="filter__input"
      onChange={[Function]}
      placeholder="Search by Title"
      type="text"
      value=""
    />
    <select
      className="filter__selector"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        All
      </option>
      <option>
        SeaFood
      </option>
      <option>
        Chicken
      </option>
      <option>
        Dessert
      </option>
      <option>
        Salad
      </option>
      <option>
        Beef
      </option>
    </select>
    <button
      className="filter__button"
      onClick={[Function]}
      type="button"
    >
      Clear Filters
    </button>
  </div>
</div>
`;
